UserProfile detail 오류 수정 && (Email, Nickname) 중복 확인 API 추가 && 친구요청 status code 변경

cafitac opened the pull request:
### member/serializers.py
 - MOD: 기존 UserSerializer -> UserCreateSerializer로 이름 변경, fields에 'id' 추가
 - ADD: UserSerializer 추가
 - MOD: 기존 UserProfileSerializer -> UserProfileCreateSerializer로 이름 변경
 - ADD: UserProfileSerializer 추가
 - MOV: UserProfileCreateSerializer의 update method, UserProfileSerializer로 이동
 - MOD: FriendRequestSerializer 필드 변경 (assent, assendted_at) 제거

### member/views.py
 - ADD: UserProfileViewSet 각 action별 serializer 분리
 - MOD: UserProfile Create 작업 이후 response에 UserProfile 정보 추가, serializer.validated_data에 user_id 추가하는 작업 생성
 - ADD: ( user_email_validate, user_nickname_validate ) 추가
 - MOD: FriendRequestViewSet의 ( 이미 요청을 보냄, 이미 친구 상태 )에 대한 status code 변경

### member/urls.py
 - ADD: ( user_email_validate, user_nickname_validate ) 각각 ( email-validate, nickname-validate ) 에 연결

hanseongd commented on scheduler/member/serializers.py:
```diff
@@ -47,6 +45,20 @@ def update(self, instance, validated_data):
+    def create(self, validated_data):
+        user_data = validated_data.pop('user')
+        user = UserCreateSerializer.create(UserCreateSerializer(), validated_data=user_data)
+        user_profile = UserProfile.objects.create(user=user, birth=validated_data.pop('birth'), language=validated_data.pop('language'))
```
user_profile 을 리턴하는데 validated_data에서 값을 가져올 때 왜 pop을 통해서 가져오는지 ?

hanseongd commented on scheduler/member/views.py:
```diff
@@ -17,7 +17,13 @@ class UserProfileViewSet(viewsets.GenericViewSet,
                          mixins.DestroyModelMixin):
     queryset = UserProfile.objects.all()
-    serializer_class = UserProfileSerializer
+    serializer_class = UserProfileCreateSerializer
```
밑 get_serializer_class 가 기능을 대체하고 있는데 불필요한 부분으로 보임

hanseongd commented on scheduler/member/views.py:
```diff
@@ -17,7 +17,13 @@ class UserProfileViewSet(viewsets.GenericViewSet,
+
+    def get_serializer_class(self):
+        if self.action == 'list':
+            return UserProfileCreateSerializer
```
action이 list일때 CreateSerializer를 return 한다?

hanseongd commented on scheduler/member/views.py:
```diff
@@ -85,6 +92,50 @@ def search(self, request):
+        serializer = self.get_serializer(data=request.data)
+
+        if not serializer.is_valid():
+            if serializer.errors['email'][0] == "Enter a valid email address.":
```
serializer 내부에서 error_message 재정의 할 수 있는거로 아는데 수정해보기
혹은 model에서 error message 정의

hanseongd commented on scheduler/member/views.py:
```diff
@@ -85,6 +92,50 @@ def search(self, request):
             })
 
 
+class UserEmailValidateViewSet(viewsets.GenericViewSet, mixins.CreateModelMixin):
```
인증만을 위한 함수인 것으로 보이는데 FBV로 만드는게 더 간단해보여

cafitac replied:
제가 router에 이 뷰를 등록하고 싶어서 CBV로 하려고 했던건데 그러면 그냥 ViewSet을 쓰지 말고 APIView를 쓸까요?

hanseongd replied:
나중에 기능이 더 확장될 일이 있는 함수라면 class로 짜는게 맞지만 그게 아니라면 이건 FBV가 맞을거 같아

hanseongd commented on scheduler/member/views.py:
```diff
@@ -93,53 +125,60 @@ class FriendRequestViewSet(viewsets.GenericViewSet,
+            }, status=status.HTTP_200_OK)
+
+
+def jwt_response_payload_handler(token, user=None, request=None):
```
jwt 관련 함수는 views.py에 같이 두는 거보다 따로 jwt 관련 함수들 작성하는 파일로 관리하는게 나을거 같아

hanseongd commented on scheduler/member/views.py:
```diff
@@ -93,53 +125,49 @@ class FriendRequestViewSet(viewsets.GenericViewSet,
+            friend_request.assented_at = timezone.now()
+            friend_request.save()
+
+            request_user = UserProfile.objects.get(user__id=int(request.data['request_user']))
```
request_user_profile

hanseongd commented on scheduler/member/views.py:
```diff
@@ -93,53 +125,49 @@ class FriendRequestViewSet(viewsets.GenericViewSet,
+            request_user = UserProfile.objects.get(user__id=int(request.data['request_user']))
+            response_user = UserProfile.objects.get(user_id=int(request.data['response_user']))
+
+            req_user = User.objects.get(id=request_user.user_id)
```
request_user = request_user_profile.user

hanseongd commented on scheduler/member/views.py:
```diff
@@ -93,53 +125,49 @@ class FriendRequestViewSet(viewsets.GenericViewSet,
+            friend_request.save()
+
+            request_user_profile = UserProfile.objects.get(user__id=int(request.data['request_user']))
+            response_user_profile = UserProfile.objects.get(user_id=int(request.data['response_user']))
```
get(user_id= .. => get(user__id=

hanseongd commented on scheduler/member/views.py:
```diff
@@ -93,53 +125,49 @@ class FriendRequestViewSet(viewsets.GenericViewSet,
+            request_user_profile = UserProfile.objects.get(user__id=int(request.data['request_user']))
+            response_user_profile = UserProfile.objects.get(user_id=int(request.data['response_user']))
+
+            req_user = User.objects.get(id=request_user_profile.user_id)
```
req_user = User.objects.get(id=request_user_profile.user_id) ==> request_user = request_user_profile.user

Commit:
UserProfile detail 오류 수정 && (Email, Nickname) 중복 확인 API 추가 && 친구요청 status code 변경

member/serializers.py
 - MOD: 기존 UserSerializer -> UserCreateSerializer로 이름 변경, fields에 'id' 추가
 - ADD: UserSerializer 추가
 - MOD: 기존 UserProfileSerializer -> UserProfileCreateSerializer로 이름 변경
 - ADD: UserProfileSerializer 추가
 - MOV: UserProfileCreateSerializer의 update method, UserProfileSerializer로 이동
 - MOD: FriendRequestSerializer 필드 변경 (assent, assendted_at) 제거
member/views.py
 - ADD: UserProfileViewSet 각 action별 serializer 분리
 - MOD: UserProfile Create 작업 이후 response에 UserProfile 정보 추가, serializer.validated_data에 user_id 추가하는 작업 생성
 - ADD: ( user_email_validate, user_nickname_validate ) 추가
 - MOD: FriendRequestViewSet의 ( 이미 요청을 보냄, 이미 친구 상태 )에 대한 status code 변경
 - MOD: FriendRequest 요청 로직 변경
member/urls.py
 - ADD: ( user_email_validate, user_nickname_validate ) 각각 ( email-validate, nickname-validate ) 에 연결
scheduler/settings.py
 - ADD: JWT_AUTH의 JWT_RESPONSE_PAYLOAD_HANDLER을 jwt_response_payload_handler로 연결

ADD: member/jwt_response.py 생성
 - ADD: jwt_response_payload_handler(jwt-token response customize 뷰) 생성
